mediapipe/graphs/instantmotiontracking/subgraphs/calculators/tracked_anchor_manager_calculator.cc

```diff
@@ -96,70 +97,70 @@ REGISTER_CALCULATOR(TrackedAnchorManagerCalculator);
-        // Check if anchor was repositioned by user, and reset tracking box
-        if(previous_anchor.x != current_anchor.x || previous_anchor.y != current_anchor.y) {
-          TimedBoxProto* box = pos_boxes->add_box();
-          box->set_left(current_anchor.x - kBoxEdgeSize * 0.5f);
-          box->set_right(current_anchor.x + kBoxEdgeSize * 0.5f);
-          box->set_top(current_anchor.y - kBoxEdgeSize * 0.5f);
-          box->set_bottom(current_anchor.y + kBoxEdgeSize * 0.5f);
-          box->set_id(current_anchor.sticker_id);
-          box->set_time_msec(cc->InputTimestamp().Microseconds() / kUsToMs);
-          current_anchor.z = 1.0; // Default value for normalized z (scale factor)
-        }
-        // If anchor was not repositioned, update the location from the tracking system if associated box exists
-        else if (cc->Inputs().HasTag(kBoxesInputTag)) {
-          const TimedBoxProtoList box_list = cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>();
-          for(const auto& box : box_list.box())
-          {
-            if(box.id() == current_anchor.sticker_id) {
-              // Get center x normalized coordinate [0.0-1.0]
-              current_anchor.x = (box.left() + box.right()) * 0.5f;
-
-              // Get center y normalized coordinate [0.0-1.0]
-              current_anchor.y = (box.top() + box.bottom()) * 0.5f;
-
-              // Get center z coordinate [z starts at normalized 1.0 and scales inversely with box-width]
-              // TODO: Look into issues with uniform scaling on x-axis and y-axis
-              current_anchor.z = kBoxEdgeSize / (box.right() - box.left());
-            }
-          }
-        }
-        tracked_scaled_anchor_data.emplace_back(current_anchor);
-        break;
-      }
-    }
-  }
-
-  mediapipe::Timestamp timestamp = cc->InputTimestamp();
-  // Delete all boxes that are artifacts from deleted anchors
-  for(const TimedBoxProto& box : cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>().box()) {
+  for(Anchor anchor : current_anchor_data) {
+    // Check if anchor is currently being tracked
     bool anchor_is_being_tracked = false;
-    for(Anchor point : tracked_scaled_anchor_data) {
-      if(box.id() == point.sticker_id) {
+    for(const TimedBoxProto& box : cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>().box()) {
+      if(box.id() == anchor.sticker_id) {
         anchor_is_being_tracked = true;
         break;
       }
     }
-    if(!anchor_is_being_tracked) {
-      // TODO: Pass in vector of boxes to prevent breaking of terms from SetOffset
-      cc->Outputs().Tag(kCancelTag).AddPacket(MakePacket<int>(box.id()).At(timestamp++));
+    // Check if initial anchor position has changed
+    bool anchor_replaced = false;
+    for(const Anchor &previous_anchor : previous_iteration_anchor_data_) {
+      if(previous_anchor.sticker_id == anchor.sticker_id) {
+        if(previous_anchor.x != anchor.x || previous_anchor.y != anchor.y) {
```

Comment:
So if the sticker_id *doesn't* match, then we don't replace the anchor, regardless of previous_anchor.x/y?  That seems unintuitive-- if this is intentional, we should have some comments for explanation.

(Also, if somehow the user manages to choose the same precise x/y location-- maybe from the Android app having some default setting-- then the anchor .z value wouldn't be updated.  Probably not a concern given how unlikely that is to occur with our current app setup, but I think from an architectural standpoint, it's generally nicer to have things be more event-driven or signal-driven, and not have to code special behavior into the values themselves).